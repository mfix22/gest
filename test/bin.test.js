const path = require('path')
const execa = require('execa')
const stripAnsi = require('strip-ansi')

function run(command) {
  return execa.stdout('node', [path.resolve(__dirname, '..', 'bin', 'gest'), command])
}

test.each(['--inspect', '-I', '{ test }'])(`success cases %s`, async flag => {
  expect(stripAnsi(await run(flag))).toMatchSnapshot()
})

test.each(['{ _test }'])(`failures case: %s`, async flag => {
  try {
    await run(flag)
  } catch (e) {
    expect(stripAnsi(e.stdout)).toMatchSnapshot()
  }
})
