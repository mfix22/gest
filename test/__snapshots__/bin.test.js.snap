// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`failures case: { _test } 1`] = `
"
Errors
- Cannot query field \\"_test\\" on type \\"Query\\". Did you mean \\"test\\"?

"
`;

exports[`success cases { test } 1`] = `
"
{
  \\"data\\": {
    \\"test\\": \\"success!\\"
  }
}
"
`;

exports[`success cases --inspect 1`] = `
"
type Mutation {
  updateUser(email: String, phone: Int, isValid: Boolean): String
  setLogLevel(input: Boolean!): Float
}

type Query {
  test: String
}
"
`;

exports[`success cases -I 1`] = `
"
type Mutation {
  updateUser(email: String, phone: Int, isValid: Boolean): String
  setLogLevel(input: Boolean!): Float
}

type Query {
  test: String
}
"
`;
